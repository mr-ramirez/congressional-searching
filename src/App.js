import 'isomorphic-fetch';
import React, { Component } from 'react';
import fetch, { Headers } from 'node-fetch';
import logo from './logo.png';
import './App.css';
import SearchForm from './components/Search/SearchForm';
import Member from './models/member';

class App extends Component {
  constructor() {
    super();

    this.state = {
      members: [],
    };
  }

  UNSAFE_componentWillMount() {
    const session = 115; // 115th congressional session
    const chamber = 'senate'; // or 'house'

    fetch(`https://api.propublica.org/congress/v1/${session}/${chamber}/members.json`, {
      headers: new Headers({
        'X-API-Key': 'd0ywBucVrXRlMQhENZxRtL3O7NPgtou2mwnLARTr',
      }),
    })
      .then((res) => res.json())
      .then((json) => json.results[0].members)
      .then((members) => {
        this.setState({
          members: members.map(member => new Member(member)),
        });
      });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">React Programming Exercise</h1>
        </header>
        <section className="container">
          {/*
            Your app should render this part of the page.
          */}
          <SearchForm results={this.state.members} />
        </section>
      </div>
    );
  }
}

export default App;
