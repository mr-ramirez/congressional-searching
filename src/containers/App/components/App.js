import { connect } from 'react-redux';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import '../App.css';
import { loadAllMembers } from '../actions';
import Header from '../../../components/Header';
import Search from '../../Search/components/Search';
import Spinner from '../../../components/Spinner';

class App extends Component {
  constructor() {
    super();

    this.state = {
      members: [],
    };
  }

  UNSAFE_componentWillMount() {
    const { chamber, congress } = this.props.search;

    this.props.loadAllMembers({ chamber, congress });
  }

  getLoadingSpinner = () => {
    const { loading } = this.props.app;

    if (loading) {
      return (<Spinner />);
    }
    return null;
  }
  

  render() {
    return (
      <div className="App">
        <Header />

        <section className="container">
          <div className="row mt-4 mb-2">
            <div className="col-lg-12 text-left">
              <h3 className="lead">Members of the Congress Search Engine</h3>
            </div>
          </div>

          { this.getLoadingSpinner() }
          {
            this.props.app.members.length > 0 ?
              <Search />
            :
              null
          }
        </section>
      </div>
    );
  }
}

App.propTypes = {
  app: PropTypes.object.isRequired,
  search: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    ...state,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    loadAllMembers: ({ chamber, congress }) =>
      dispatch(loadAllMembers({ chamber, congress })),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
