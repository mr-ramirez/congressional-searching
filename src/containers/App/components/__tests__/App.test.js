import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

jest.mock('../../../Search/components/SearchForm', () => 'SearchForm');
jest.mock('../../../../components/Spinner', () => 'Spinner');

describe('App.js', () => {
  const div = document.createElement('div');

  const mockedStore = {
    dispatch: jest.fn(),
    getState: jest.fn(),
    subscribe: jest.fn(),
  };

  const mockedPropsSearch = {
    chamber: 'house',
    congress: 100,
  };

  const mockedPropsApp = {
    loading: true,
    members: [{
      id: 10,
    }],
  };

  describe('WHEN property <loading> is true and <members> is not empty', () => {
    it('THEN it should render with a spinner and the search component', () => {
      const component = renderer.create(
        <App
          app={mockedPropsApp}
          store={mockedStore}
          search={mockedPropsSearch} />,
      );
      const tree = component.toJSON();
      
      expect(tree).toMatchSnapshot();
      expect(tree.children[1].children[0].type).toEqual('spinner');
      expect(tree.children[1].children[1].type).toEqual('SearchForm');
    });
  });
});
