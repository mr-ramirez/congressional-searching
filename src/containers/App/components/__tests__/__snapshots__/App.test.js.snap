// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js WHEN property <loading> is true and <members> is not empty THEN it should render with a spinner and the search component 1`] = `
<div
  className="App"
>
  <header
    className="Header_container"
  >
    <img
      alt="logo"
      className="Header_logo"
      src="logo.png"
    />
    <h1
      className="Header_title"
    >
      React Programming Exercise
    </h1>
  </header>
  <section
    className="container"
  >
    <div
      className="row mt-4 mb-2"
    >
      <div
        className="col-lg-12 text-left"
      >
        <h3
          className="lead"
        >
          Members of the Congress Search Engine
        </h3>
      </div>
    </div>
    <Spinner />
    <Search />
  </section>
</div>
`;
